### PR TITLE
initial implementation of access_tokens table and logout

### PR DESCRIPTION
**Pull request by derekleverenz**
This addresses https://github.com/opeolluwa/raccoon/issues/163

Not yet ready for merge, remaining work:

- [ ] manual end to end testing
- [ ] address questions & todos

The questions I have are
1. should the `token` column in `acesss_tokens` have an index? i think it should unless its intended to be kept small
2. does the `access_tokens` table need to h ave anything other than the token itself, and does it need to store tokens when they are created, or can it store only when logged out?

### CODE REVIEW

**Comment by opeolluwa** [comment]
> This addresses #163
> 
> Not yet ready for merge, remaining work:
> 
> * [ ]  manual end to end testing
> * [ ]  address questions & todos
> 
> The questions I have are
> 
> 1. should the `token` column in `acesss_tokens` have an index? i think it should unless it's intended to be kept small
> 2. does the `access_tokens` table need to have anything other than the token itself, and does it need to store tokens when they are created, or can it store only when logged out?

@derekleverenz  First off, well done for the good work,  I appreciate you ❤️

Regards the questions you're asked. 
I made a review the algorithm I wrote earlier. 

I believe it's best to persist the token when the user logs out only. 

And for the second question,  let's have the table also store the logout time stamp. This hook will later be used to remove expired and invalidated tokens. 

In summary, 
1. Create the token validation table having the `token` field and the `last_valid_at` - timestamp the token was saved 
2. When a user logs out, save the token. 
4. Update the [token validation method](https://github.com/opeolluwa/raccoon/blob/master/src/utils/jwt.rs#L57) to check if the token is in the `access_token` table, if it is, this would imply the token has been invalidated by the user. Return an `Unauthorized error` telling the user to log in again. 

Also, Index would be great to achieve `step 4` defined above with relatively low latency.

**Review by opeolluwa — APPROVED** [review]
